replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/legacy/LegacyDataConverter.java

```diff
@@ -0,0 +1,248 @@
+
+  private final ReplicatorConfigManager newConfigManager;
+
+  private final ReplicationPersistentStore persistentStore;
+
+  private final Security security;
+
+  public LegacyDataConverter(
+      CatalogFramework framework,
+      FilterBuilder filterBuilder,
+      MetacardHelper helper,
+      SiteManager siteManager,
+      ReplicatorConfigManager newConfigManager,
+      ReplicationPersistentStore persistentStore) {
+    this(
+        framework,
+        filterBuilder,
+        helper,
+        siteManager,
+        newConfigManager,
+        persistentStore,
+        Security.getInstance());
+  }
+
+  @VisibleForTesting
+  LegacyDataConverter(
+      CatalogFramework framework,
+      FilterBuilder filterBuilder,
+      MetacardHelper helper,
+      SiteManager siteManager,
+      ReplicatorConfigManager newConfigManager,
+      ReplicationPersistentStore persistentStore,
+      Security security) {
+    this.framework = framework;
+    this.filterBuilder = filterBuilder;
+    this.helper = helper;
+    this.siteManager = siteManager;
+    this.newConfigManager = newConfigManager;
+    this.persistentStore = persistentStore;
+    this.security = security;
+  }
+
+  public void init() {
+    Failsafe.with(
+            new RetryPolicy()
+                .withBackoff(3, 60, TimeUnit.SECONDS)
+                .withMaxDuration(10, TimeUnit.MINUTES))
+        .onFailedAttempt(
+            throwable -> LOGGER.debug("failed to upgrade legacy configs, will retry", throwable))
+        .onFailure(
+            throwable ->
+                LOGGER.debug("failed to upgrade legacy configs, restart to try again.", throwable))
+        .run(this::convertConfigs);
+  }
+
+  private void convertConfigs() {
+    // convert sites first
+    persistentStore.objects(OldSite.class).forEach(this::convert020Site);
+
+    security.runAsAdmin(
+        () -> {
+          try {
+            security.runWithSubjectOrElevate(
+                () -> {
+                  getAllConfigs()
+                      .forEach(
+                          config -> {
+                            newConfigManager.save(config);
+                            removeConfig(config);
+                          });
+                  return null;
+                });
+          } catch (SecurityServiceException | InvocationTargetException e) {
+            LOGGER.warn(
+                "Error occured while trying to transform catalog saved metacards to persistent store saved configs. Legacy configs may not be accessible.",
+                e);
+          }
+          return null;
+        });
+  }
+
+  List<ReplicatorConfig> getAllConfigs() {
+    return helper.getTypeForFilter(
+        filterBuilder.attribute(Metacard.TAGS).is().like().text(ReplicationConfig.METACARD_TAG),
+        this::getConfigFromMetacard);
+  }
+
+  void removeConfig(ReplicatorConfig replicationConfig) {
+    try {
+      framework.delete(new DeleteRequestImpl(replicationConfig.getId()));
```

Comment:
❓ Should we go straight to the provider to avoid versioning off the config?

Reply:
I'm not sure if we care about that. we were using the framework before so I assume this would still be what we want to use.